bin/pycbc_compress_bank

```diff
@@ -0,0 +1,199 @@
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+__description__ = \
+""""Loads a bank of waveforms and compresses them using the specified
+compression algorithm. The resulting compressed waveforms are saved to an
+hdf file."""
+
+import argparse
+import numpy
+import h5py
+import logging
+import pycbc
+from pycbc.waveform import compress
+from pycbc import waveform
+from pycbc.types import FrequencySeries
+from pycbc import pnutils
+from pycbc import filter
+
+parser = argparse.ArgumentParser(description=__description__)
+parser.add_argument("--bank-file", type=str,
+                help="Bank hdf file to load.")
+parser.add_argument("--output", type=str,
+                help="The hdf file to save the templates and compressed "
+                     "waveforms to.")
+parser.add_argument("--low-frequency-cutoff", type=float, required=True,
+                help="The low frequency cutoff to use for generating "
+                     "the waveforms (Hz)")
+parser.add_argument("--approximant", type=str,
+                help="The approximant to generate the waveforms with. If "
+                     "none provided, will attempt to retrieve the approximant "
+                     "from the bank file.")
+parser.add_argument("--sample-rate", type=int, required=True,
+                help="Half this value sets the maximum frequency of the "
+                     "compressed waveforms.")
+parser.add_argument("--segment-length", type=float, required=True,
+                help="The inverse of this sets the delta-f used for "
+                     "generating waveforms prior to compressing.")
+parser.add_argument("--tmplt-index", nargs=2, type=int, default=None,
+                help="Only generate compressed waveforms for the given "
+                     "indices in the bank file. Must provide both a start "
+                     "and a stop index. Default is to compress all of the "
+                     "templates in the bank file.")
+parser.add_argument("--compression-algorithm", type=str, default="mchirp",
+                help="The compression algorithm to use for selecting "
+                     "frequency points. Options are %s. Default is mchirp." %(
+                     ', '.join(compress.compression_algorithms.keys())))
```

Comment:
Hi Collin, you can use ArgumentParser's `choices` kwarg here, see other PyCBC scripts for examples.